### PR TITLE
Handle Boolean literals as enum raw values. 

### PR DESCRIPTION
**Pull request by DougGregor**
Fixes rdar://problem/55384273.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
Is the depth map stuff supposed to be attached?

**Comment by DougGregor** [comment]
> Is the depth map stuff supposed to be attached?

I got lazy, sorry; it's also in the more purposeful https://github.com/apple/swift/pull/28276

File: test/decl/enum/bool_raw_value.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+		self = value != 0
+	}
+}
+
+enum IsDefinitelyRecursive : Bool, Equatable, Hashable {
+  case recursive=false
+}
+
+// expected-error@+1{{'IsRecursive' declares raw type 'Bool', but does not conform to RawRepresentable and conformance could not be synthesized}}
+enum IsRecursive : Bool, Equatable, Hashable {
```
Comment (CodaFi) [inline test/decl/enum/bool_raw_value.swift]: Some more tests: an @objc enum (and if we allow this, an IRGen test), an enum with an invalid raw value expression.

Comment (DougGregor) [reply test/decl/enum/bool_raw_value.swift]: `@objc` enums are banned; I've added more tests

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Pull request merged**